sdk/storage/azblob/blockblob/client.go

```diff
@@ -169,6 +171,15 @@ func (bb *Client) StageBlock(ctx context.Context, base64BlockID string, body io.
 
 	opts, leaseAccessConditions, cpkInfo, cpkScopeInfo := options.format()
 
+	if options.TransactionalContentCRC64 == 0 && options.TransactionalValidationOption != hashing.StorageTransferValidationOptionNone {
+		body, err = hashing.NewHashingReadWrapper(body, options.TransactionalValidationOption)
+
+		if options.TransactionalValidationOption&hashing.StorageTransferValidationOptionCRC64 == hashing.StorageTransferValidationOptionCRC64 {
```

Comment:
This same check is done inside NewHashingReadWrapper() as well. In future if more options of validation come then this will be duplicate.

Reply:
This check is done to prevent writing an empty header.